bindings/c/test/apitester/TesterApiWorkload.cpp

```diff
@@ -152,6 +152,21 @@ fdb::Key ApiWorkload::randomKey(double existingKeyRatio, std::optional<int> tena
 	}
 }
 
+fdb::KeyRange ApiWorkload::randomKeyRange() {
```

Comment:
Do we test behavior on empty key ranges? If they are considered as invalid then some scripted unit tests would suffice to check if the API reports some meaningful error (rather than crashing or causing some undefined behavior)

Reply:
I have the same concern: why is the empty range invalid?

Reply:
Yeah it's not invalid as much as it is a no-op, so it's not properly testing what we want to test. The more pressing issue was a bug where the test's validation assumed the range wasn't empty.
I agree that we should validate behavior on an empty range though. What I'd propose is to add empty-range testing like you describe to the TesterBlobGranuleErrorsWorkload, and then leave this as-is in the TesterBlobGranuleCorrectnessWorkload

Reply:
Yes, that would work. 

Rename the method to randomNonEmptyKeyRange(), so that for somebody who decides to use it, it is clear that the method does not cover all possible inputs for key ranges.

Reply:
Done